modules/cli/src/main/scala/scala/cli/internal/ProcUtil.scala

```diff
@@ -29,4 +31,12 @@ object ProcUtil {
     usesSh
   }
 
+  def downloadFile(url: String): String = {
+    val connection = new URL(url).openConnection.asInstanceOf[HttpURLConnection]
+    connection.setRequestMethod("GET")
+    val inputStream = connection.getInputStream
+    val br          = new BufferedReader(new InputStreamReader(inputStream))
+    Stream.continually(br.readLine()).takeWhile(_ != null).mkString("\n")
```

Comment:
Not sure we should use a `BufferedReader` here… Let's just read it all in a `ByteArrayOutputStream`, and convert to a string in one go. Also, the connection isn't closed it seems. And we shouldn't need to set the request method to GET (that should be the default, AFAIK). Something like this should do:
```suggestion
    val buffer = new ByteArrayOutputStream
    val data   = Array.ofDim[Byte](16384)
    var inputStream: InputStream = null
    try {
      inputStream = new URL(url).openStream()
      var read = 0
      while ({
        read = inputStream.read(data, 0, data.length)
        read != -1
      })
        buffer.write(data, 0, read)
    }
    finally {
      if (inputStream != null)
        inputStream.close()
    }
    new String(buffer.toByteArray, StandardCharsets.UTF_8)
```

Reply:
(Recent JVMs might even have helpers to convert an `InputStream` to an `Array[Byte]`…)

Reply:
Yeah
```suggestion
    var inputStream: InputStream = null
    val data =
      try {
        inputStream = new URL(url).openStream()
        inputStream. readAllBytes()
      }
      finally {
        if (inputStream != null)
          inputStream.close()
      }
    new String(data, StandardCharsets.UTF_8)
```